srml/support/src/dispatch.rs

```diff
@@ -67,37 +67,62 @@ pub trait Parameter: Codec + Clone + Eq {}
+/// Declares a `Module` struct and a `Call` enum, which implements the dispatch logic.
 ///
-/// Usually used as follows:
+/// ## Example
 ///
+/// ```rust,ignore
 /// decl_module! {
-///		pub struct Module<T: Trait> for enum Call where origin: T::Origin
-///{}
-///	}
+/// 	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+/// 		fn foobar(origin, ...) -> Result {
+///				...
+/// 			OK(())
+/// 		}
+///			pub fn foo(origin) -> Result {...}  // public function easily accessible to other modules
+///			fn bar(origin) {...}				// macro expands to return `Result`
+///			fn baz() -> Result {...}			// privileged function
+///		}
+/// }
+/// ```
+///
+/// *Where `...` indicates truncated code for example brevity.*
+///
+/// Function declaration is set with the header where:
+///
+/// * `Module`: The struct generated by the macro, with type `Trait`.
+/// * `Call`: The enum generated for every module, which implements [`Callable`](./dispatch/trait.Callable.html).
+/// * `origin`: Alias of `T::Origin`, declared by the [`impl_outer_origin!`](./macro.impl_outer_origin.html) macro.
+/// * `Result`: The expected return type from module functions.
+///
+/// Function declaration variations include:
+///
+/// * [Optional] Public functions: `pub` functions are easily accessible to other modules, e.g. `foo`.
+/// * [Optional] Omit returning `Result`: the macro auto-expands functions to return a `Result` type, e.g. `bar`.
+/// * [Optional] Privileged calls: if the `origin` param is omitted, the macro assumes the `origin`
+///   to be `Root`, e.g. `baz`.
+///
+/// ## Special Functions
+///
+/// Special functions are executed once per block and have reserved function signatures.
 ///
-/// where "Trait" is a trait describing this module's requirements for the Runtime type.
-/// T::Origin is declared using a impl_outer_origin! per-module macro (which is generated by the
-/// construct_runtime! macro) and automatically includes all the modules that are used by the
-/// runtime (alongside with a variant called "system").
+/// * `deposit_event`: Emits [events](https://docs.substrate.dev/docs/event-enum) for the module.
+/// [Optional]: `deposit_event<T>` if the event contains generics.
 ///
-/// A runtime module is a collection of functions unified by a common problem domain and certain
-/// shared types. The "functions" do not actually return values (see Dispatchable) and are only
-/// used for side effects.
+/// The following functions also take in one optional parameter `n: T::BlockNumber`:
 ///
-/// For every module an associated enum (usually "Call") is generated with every variant
-/// corresponding to a function of the module. This enum implements Callable and thus its values
-/// can be used as an extrinsic's payload.
+/// * `on_initialize`: Executes at the beginning of a block.
+/// * `on_finalize`: Executes at the end of a block.
+/// * `offchain_worker`: Starts at the beginning of a block and produces extrinsics for the next block.
```

Comment:
Does it actually produce extrinsics for the "next block" or "some future block". I thought we allow for processes that take multiple block times to produce.

Reply:
correct correction - changing it to specify "some future block "